nncf/onnx/graph/nncf_graph_builder.py

```diff
@@ -348,6 +349,7 @@ def create_nncf_graph(onnx_model: onnx.ModelProto) -> NNCFGraph:
         """
         onnx_model = GraphConverter._replace_empty_node_name(onnx_model)
         onnx_model = onnx.shape_inference.infer_shapes(onnx_model)
+        onnx_model = onnxoptimizer.optimize(onnx_model, ["eliminate_nop_cast"])
```

Comment:
The pass should be applied after `onnx.shape_inference.infer_shapes()` call. Otherwise, not all no-op `Cast` nodes will be found."